flamingo_tools/segmentation/postprocessing.py

```diff
@@ -319,6 +319,31 @@ def downscaled_centroids(
     return new_array
 
 
+def graph_connected_components(coords, min_edge_distance, min_component_length):
+    graph = nx.Graph()
+    for num, pos in coords.items():
+        graph.add_node(num, pos=pos)
+
+    # create edges between points whose distance is less than threshold min_edge_distance
+    for i in coords:
+        for j in coords:
```

Comment:
This looks wrong. Shouldn't `i, j` be indices into the respective array or list?
So 
```for i in range(len(coords))```
etc. ?